### PR TITLE
use pyproject.toml instead of setup.py

### PR DESCRIPTION
**Pull request by petrjasek**
also switched from black/flake8 to ruff

### CODE REVIEW

**Review by MarkLark86 — APPROVED** [review]
Why the change from `black` to `ruff` (performance)? We should probably use pyproject.toml in other repos too

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+dependencies = [
+    "arrow>=0.4.2",
+    "ciso8601>=1.0.2,<3",
+    "pytz>=2015.4",
+    "elasticsearch>=7.0,<8.0",
+]
+license = {file = "LICENSE"}
+authors = [
+    {name = "Petr Jasek"},
+    {name = "Mark Lark"},
```
Comment (MarkLark86) [inline pyproject.toml:21]: This should be `Mark Pittaway`

**Comment by petrjasek** [comment]
> Why the change from `black` to `ruff` (performance)?

ruff handles both formatting and linting, so it can replace also flake8

>  We should probably use pyproject.toml in other repos too

yep I plan to do it but there is this babel issue https://github.com/python-babel/babel/issues/777

File: pyproject.toml (hunk 1 above)
Comment (petrjasek) [reply pyproject.toml:21]: :man_facepalming: sorry

**Pull request merged**